libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -366,12 +395,26 @@ class MapboxNavigation @VisibleForTesting internal constructor(
             historyRecorder.fileDirectory(),
             logger,
             navigationOptions.accessToken ?: "",
+            if (moduleRouter.isInternalImplementation()) {
+                nativeRouter
+            } else {
+                RouterInterfaceDelegate(moduleRouter)
+            },
```

Comment:
Why can't we always wrap routes? Is that because of performance?

Reply:
you mean `Router`, right? SDK's Router already depends on Native Router, so it will cycle dependency. When we migrate Reroute and Refresh to Native, we should start to depend on the MapboxNavigation interface only. 
see https://github.com/mapbox/mapbox-navigation-android/issues/5501

Reply:
Note: add marker interface